src/pages/api/applications/nonprofit.ts

```diff
@@ -1,22 +1,43 @@
+import { JTDDataType } from "ajv/lib/types/jtd-schema";
 import { Types } from "mongoose";
 import { nonprofitCreateApplication } from "server/mongodb/actions/Application";
-import APIWrapper from "server/utils/APIWrapper";
-import { NonprofitCreateApplication, Role } from "src/utils/types";
+import { APIWrapper } from "server/utils/APIWrapper";
+import { InternalRequest, Role } from "src/utils/types";
+
+const POST_BODY_SCHEMA = {
+  properties: {
+    projectId: {
+      type: "string",
+    },
+    applicationCreate: {
+      optionalProperties: {
+        aboutQ1: { type: "string" },
+        aboutQ2: { type: "string" },
+        aboutQ3: { type: "string" },
+        aboutQ4: { type: "string" },
+        needsQ1: { type: "string" },
+        needsQ2: { type: "string" },
+        needsQ3: { type: "string" },
+        needsQ4: { type: "string" },
+        needsQ5: { type: "string" },
+      },
+    },
+  },
+} as const;
 
 export default APIWrapper({
   POST: {
     config: {
       requireSession: true,
       roles: [Role.NONPROFIT_ADMIN],
+      bodySchema: POST_BODY_SCHEMA,
     },
-    handler: async (req) => {
-      const applicationCreate = req.body
-        .applicationCreate as NonprofitCreateApplication;
-      const projectId = req.body.projectId as string;
-
+    handler: async (
+      req: InternalRequest<undefined, JTDDataType<typeof POST_BODY_SCHEMA>>
+    ) => {
```

Comment:
Type inference does not play friendly here. You need to specify the type of the request in order for it to play friendly with the compiler. This can be done here or it can be done on the entire route, e.g. `as Route<QUERY_PARAMS_TYPE, BODY_TYPE, RETURN_TYPE`